compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -620,7 +620,7 @@ private BLangConstantValue readConstLiteralValue(DataInputStream dataInStream) t
             case TypeTags.DECIMAL:
                 return new BLangConstantValue(getStringCPEntryValue(dataInStream), symTable.decimalType);
             case TypeTags.BOOLEAN:
-                return new BLangConstantValue(dataInStream.readByte() == 1, symTable.booleanType);
```

Comment:
Is it ok to change this? We'll have to inform llvm team to change there code.

Reply:
We are writing boolean and reading on CP table. So doing the same in other places also will be consistent which is why the above change. Also, it makes sense to write and read boolean values for boolean constants than writing byte (even though we are doing the same thing writing 1 or 0 byte effectively)